activestorage/app/models/active_storage/blob/representable.rb

```diff
@@ -112,10 +112,10 @@ def default_variant_format
     end
 
     def format
-      if filename.extension.present? && MiniMime.lookup_by_extension(filename.extension)&.content_type == content_type
+      if filename.extension.present? && Marcel::MimeType.for(extension: filename.extension) == content_type
         filename.extension
       else
-        MiniMime.lookup_by_content_type(content_type)&.extension
+        Marcel::Magic.new(content_type).extensions.first
```

Comment:
We pick first extension from Marcel. MiniMime returns singular extension based on content_type.

Reply:
```ruby
3.2.1 :008 > Marcel::Magic.new("").extensions.first
 => nil
3.2.1 :009 > Marcel::Magic.new("application/octet-stream").extensions.first
 => "bin"
```